src/tikzdocument.jl

```diff
@@ -22,43 +22,57 @@ push_preamble!(td::TikzDocument, v) = (push!(td.preamble, v); td)
 # Output #
 ##########
 
-function save(filename::String, td::TikzDocument; include_preamble::Bool = true,
-                                                  latex_engine = latexengine(),
-                                                  buildflags = vcat(DEFAULT_FLAGS, CUSTOM_FLAGS),
-                                                  dpi = 150)
-    file_ending = split(basename(filename), '.')[end]
-    filename_stripped = filename[1:end-4] # This is ugly, whatccha gonna do about it?
-    file_endings = ["tex", "svg", "pdf"]
-    HAVE_PDFTOPPM && push!(file_endings, "png")
-    if !(file_ending in file_endings)
+"""
+Extensions that make [`save`](@ref) choose a standalone `tikz` format.
+
+The saved file has no preamble, just a `tikzpicture` environment. These
+extensions should be recognized by `\includegraphics` when the
+[tikzscale](https://www.ctan.org/pkg/tikzscale) LaTeX package is used.
+"""
+const STANDALONE_TIKZ_FILEEXTS = [".tikz", ".TIKZ", ".TikZ", ".pgf", ".PGF"]
+
+function save(filename::String, td::TikzDocument;
+              include_preamble::Bool = true,
+              latex_engine = latexengine(),
+              buildflags = vcat(DEFAULT_FLAGS, CUSTOM_FLAGS),
+              dpi = 150)
+    filebase, fileext = splitext(filename)
+    if fileext == ".tex"
+        savetex(filename, td; include_preamble = include_preamble)
+    elseif fileext ∈ STANDALONE_TIKZ_FILEEXTS
+        savetex(filename, td; include_preamble = false)
+    elseif fileext == ".svg"
+        savesvg(filename, td;
+                latex_engine = latex_engine, buildflags = buildflags)
+    elseif fileext == ".pdf"
+        savepdf(filename, td;
+                latex_engine = latex_engine, buildflags = buildflags)
+    elseif HAVE_PDFTOPPM && fileext == ".png"
+        savepng(filebase, td;
```

Comment:
What is this `filebase` instead of `filename`?

Reply:
Forgot to change it, fixed it now.